src/java/org/apache/cassandra/index/sai/disk/v3/CassandraDiskAnn.java

```diff
@@ -210,6 +242,80 @@ public void close()
         }
     }
 
+    /**
+     * An iterator over {@link ScoredRowId}. This iterator is backed by a {@link SearchResult} and resumes the search
+     * when the backing {@link SearchResult} is exhausted.
+     */
+    private class ScoredRowIdIteratorDiskAnn implements ScoredRowIdIterator
```

Comment:
we could duplicate less code by using RowIdIterator on a per SearchResult basis

Reply:
I worked on deduplicating code a fair amount today. It was a bit tricky to get right, but I think I made good progress. I'll need to continue tomorrow.

Reply:
This part is ready for review.